packages/usehooks-ts/src/useTernaryDarkMode/useTernaryDarkMode.ts

```diff
@@ -26,36 +27,14 @@ interface UseTernaryDarkModeOutput {
  */
 export function useTernaryDarkMode(
   localStorageKey: string = LOCAL_STORAGE_KEY,
+  defaultValue: TernaryDarkMode = 'system',
 ): UseTernaryDarkModeOutput {
   const isDarkOS = useMediaQuery(COLOR_SCHEME_QUERY)
   const [ternaryDarkMode, setTernaryDarkMode] =
-    useLocalStorage<TernaryDarkMode>(localStorageKey, 'system')
-  const [isDarkMode, setDarkMode] = useState<boolean>(isDarkOS)
+    useLocalStorage<TernaryDarkMode>(localStorageKey, defaultValue)
+  
+  const isDarkMode = ternaryDarkMode === 'dark' || (ternaryDarkMode === 'system' && isDarkOS)
 
-  // Update darkMode if os prefers changes
-  useUpdateEffect(() => {
-    if (ternaryDarkMode === 'system') {
-      setDarkMode(isDarkOS)
-    }
-  }, [isDarkOS])
-
-  useEffect(() => {
-    switch (ternaryDarkMode) {
-      case 'light':
-        setDarkMode(false)
-        break
-      case 'system':
-        setDarkMode(isDarkOS)
-        break
-      case 'dark':
-        setDarkMode(true)
-        break
-    }
-  }, [ternaryDarkMode, isDarkOS])
```

Comment:
Have simplified this state-syncing logic as I don't believe it's necessary - it can just be derived from the above local storage & media query states.